### ISSUE TITLE
关于main_infer里的测试 img = '51aa9b8d0da890cd1d0c5029e3d89e3c.jpg'经过处理后形状为[1,6,6,512,512]的维度问题

### ISSUE THREAD

**Issue by zxy-zx** [VisionRush/DeepFakeDefenders#25]
您好，首先感谢您的开源代码，在复现的过程中，在推理的这一步报错
def main():
    
    img = '51aa9b8d0da890cd1d0c5029e3d89e3c.jpg'
    infer_api = INFER_API()
    print(infer_api.test(img))

if __name__ == '__main__':
    main()
这里图片的embedding为[1,6,6,512,512]
但是RuntimeError: Expected 3D (unbatched) or 4D (batched) input to conv2d, but got input of size: [1, 6, 6, 512, 512]
![image](https://github.com/user-attachments/assets/61b992b5-340e-46ca-aa75-72e844718c9a)
我们尝试这打印每一步变换的图片形状
![image](https://github.com/user-attachments/assets/8526795a-02dc-4ed0-9916-f503370a2dc0)
![image](https://github.com/user-attachments/assets/f022a58b-2f4e-4de4-b84a-ca172eb1598b)
请问输入到模型中的图片维度要求是什么，这里测试一张图片但是经过六种变换之后到all_data列表里面是6个[6,512,512]的张量，经过        img_tensor = torch.stack(all_data, dim=0).unsqueeze(0).cuda()之后变成了[1,6,6,512,512,]导致输入报错输入形状错误

**Comment by Fieldhunter** [comment]
您好，我猜您是否用的不是final_model这个模型，用的是单个ConvNeXt或者RepLKNet模型，又或者是别的您自己定义的模型，是这样吗？

**Comment by zxy-zx** [comment]
> 您好，我猜您是否用的不是final_model这个模型，用的是单个ConvNeXt或者RepLKNet模型，又或者是别的您自己定义的模型，是这样吗？

对的，我使用的是单个模型convnext，因为在尝试使用两个模型融合的final_model的时候gpu只有24G，显示需要27G ，所以使用了单个模型，请问推理时在哪能调小模型呢，或者如何使用单个模型

**Comment by zxy-zx** [comment]
> 您好，我猜您是否用的不是final_model这个模型，用的是单个ConvNeXt或者RepLKNet模型，又或者是别的您自己定义的模型，是这样吗？

感谢您的回答，对的，我使用的是单个模型convnext，因为在尝试使用两个模型融合的final_model的时候gpu只有24G，显示需要27G ，所以使用了单个模型，请问推理时在哪能调小模型呢，或者如何使用单个模型

**Comment by zxy-zx** [comment]
您好，我们在尝试另一个更大的gpu上已经解决该问题了，感谢您的回答，请问如何使用单个模型

**Comment by Fieldhunter** [comment]
这个5维的输入是满足final_model的处理逻辑的，如果您需要更改单个模型的推理过程，且满足5维的输入。您可以参考一下我下面写的伪代码:
```
        B, N, C, H, W = x.shape
        x = x.view(-1, C, H, W) # 变为4维

        pred = convnext(x) # 模型输出

        # 输出处理
        outputs_score = nn.functional.softmax(pred, dim=1)
        predict_score = outputs_score[:, 1]
        predict_score = predict_score.view(B, N).mean(dim=-1)

        return predict_score
```
您可以按照这个逻辑尝试一下，在模型推理前后进行相应的修改。如果仍有问题，欢迎继续提出issue。
